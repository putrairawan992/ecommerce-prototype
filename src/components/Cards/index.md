---
name: Cards
menu: Components
---

import { Playground, Props } from 'docz'
import Cards from './'
import "antd/dist/antd.less"

# Cards

This is Cards of Monggopesen

## Properties (props)

<Props of={Cards} />

## Purpose Of Props

- Type: to set the classname of the card
- urlImage: url of the image
- title : string of the product name
- price : string of the product price
- showPlayButton : boolean to show play button
- discount: to display the persentage of the discount on the flag
- discountPrice: to display the price before the discount applied
- rateValue: to set the value of rating between 0 - 5 of the average total vote
- totalVote: to show how many voters has vote the product
- totalSold: to show how many item has been sold during flash sale event
- itemLeft: to show how many item left during flash sale event

## Details

no details

## Playground

### Cards Default

<Playground>
    <Cards
        title={"Sepeda Motor"}
        urlImage={"https://s3.ap-southeast-1.amazonaws.com/bucket-monggopesen/2019-08-15T07:02:04.092Z_574ae245-06c0-49ae-b4d3-33924223a652"}
        price={25283000.00}
        />
</Playground>

### Cards Default flash sale

<Playground>
    <Cards
        title={"Sepeda Motor"}
        urlImage={"https://s3.ap-southeast-1.amazonaws.com/bucket-monggopesen/2019-08-15T07:02:04.092Z_574ae245-06c0-49ae-b4d3-33924223a652"}
        price={25283000.00}
        totalSold={12}
        itemLeft={88}
        />
</Playground>

### Cards with type="collection"

<Playground>
    <Cards
        type="collection"
        title={"Sepeda Motor"}
        urlImage={"https://s3.ap-southeast-1.amazonaws.com/bucket-monggopesen/2019-08-15T07:02:04.092Z_574ae245-06c0-49ae-b4d3-33924223a652"}
        price={25283000.00}
    />
</Playground>

### Cards with type="collection" discount price

<Playground>
    <Cards
        type="collection"
        title={"Sepeda Motor"}
        urlImage={"https://s3.ap-southeast-1.amazonaws.com/bucket-monggopesen/2019-08-15T07:02:04.092Z_574ae245-06c0-49ae-b4d3-33924223a652"}
        price={25283000.00}
        discount="60%"
        discountPrice={300000000.00}
        rateValue={2}
        totalVote={3}
    />
</Playground>

### Cards with type="large"

<Playground>
    <Cards
        type="large"
        title={"Sepeda Motor"}
        urlImage={"https://s3.ap-southeast-1.amazonaws.com/bucket-monggopesen/2019-08-15T07:02:04.092Z_574ae245-06c0-49ae-b4d3-33924223a652"}
        price={25283000.00}
    />
</Playground>

### Cards with type="small"

<Playground>
    <Cards
        type="small"
        title={"Sepeda Motor"}
        urlImage={"https://s3.ap-southeast-1.amazonaws.com/bucket-monggopesen/2019-08-15T07:02:04.092Z_574ae245-06c0-49ae-b4d3-33924223a652"}
        price={25283000.00}
    />
</Playground>
