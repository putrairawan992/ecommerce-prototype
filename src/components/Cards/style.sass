@import "../../sass/variable.sass"

.mp-card-info
    display: flex
    justify-content: space-between
    flex-direction: column
    height: 100%

.mp-card-badge
    z-index: 1
    position: absolute
    margin: 8px 0
    padding: 2px 8px
    color: $color-white
    font-weight: 300
    background-color: $color-orange-red

.mp-card-stars
    display: flex
    align-items: flex-end
    font-size: 10px
    color: $color-silver
    margin-bottom: 4px
    .ant-rate
        margin-right: 8px
        font-size: 12px

.mp-card-bar
    &__info
        font-size: 12px

.card-discount
    display: flex
    flex-direction: column

.card-discount-price
    font-size: 10px
    color: $color-border
    text-decoration: line-through

.mp-card-default
    width: 175px
    height: 295px
    border-radius: 10px
    box-shadow: 0 0 7px 0 rgba(0, 0, 0, 0.08)
    overflow: hidden
    .ant-card-body
        height: 120px
    &__title
        display: -webkit-box
        -webkit-box-orient: vertical
        -webkit-line-clamp: 2
        max-height: 38px
        overflow: hidden
        font-size: 12px
        color: $color-grey
        line-height: 16px
    &__price
        font-size: 12px
        font-weight: 500
        color: $color-teal
    &__box-image
        display: flex
        justify-content: center
        align-items: center
        height: 175px
    &__image
        max-height: 100%
        max-width: 100%

.mp-card-default:hover
    box-shadow: 0 0 8px 6px rgba(0, 0, 0, 0.06)

.mp-card-collection
    width: 175px
    height: 295px
    border-radius: 10px
    box-shadow: 0 0 4px 2px rgba(0, 0, 0, 0.08)
    overflow: hidden
    .ant-card-body
        height: 120px
    &__title
        display: -webkit-box
        -webkit-box-orient: vertical
        -webkit-line-clamp: 2
        max-height: 38px
        overflow: hidden
        font-size: 12px
        color: $color-grey
        line-height: 16px
    &__price
        font-size: 12px
        font-weight: 500
        color: $color-teal
    &__box-image
        display: flex
        justify-content: center
        align-items: center
        height: 175px
    &__image
        max-height: 100%
        max-width: 100%

.mp-card-collection:hover
    box-shadow: 0 0 10px 2px rgba(0, 0, 0, 0.2)

.mp-card-large
    width: 276px
    height: 296px
    border-radius: 8px
    box-shadow: 0 0 6px 2px rgba(0, 0, 0, 0.08)
    overflow: hidden
    .ant-card-body
        padding: 12px 24px !important
        height: 120px
    &__title
        display: -webkit-box
        -webkit-box-orient: vertical
        -webkit-line-clamp: 2
        max-height: 38px
        overflow: hidden
        font-size: 24px
        color: $color-black
        font-weight: bold
    &__description
        font-size: 18px
        font-weight: 500
        color: $color-charcoal
    &__box-image
        display: flex
        justify-content: center
        align-items: center
        height: 190px
    &__image
        max-height: 100%
        max-width: 100%

.mp-card-large:hover
    box-shadow: 0 0 8px 6px rgba(0, 0, 0, 0.06)

.mp-card-small
    width: 136px
    height: 140px
    border-radius: 4px
    box-shadow: 0 0 6px 2px rgba(0, 0, 0, 0.08)
    overflow: hidden
    line-height: 16px
    .ant-card-body
        height: 20px
    &__title
        display: -webkit-box
        -webkit-box-orient: vertical
        -webkit-line-clamp: 2
        max-height: 38px
        overflow: hidden
        font-size: 12px
        color: $color-black
        font-weight: bold
    &__description
        font-size: 10px
        font-weight: 500
        color: $color-charcoal
    &__box-image
        display: flex
        justify-content: center
        align-items: center
        height: 80px
    &__image
        max-height: 100%
        max-width: 100%

.mp-card-small:hover
    box-shadow: 0 0 8px 6px rgba(0, 0, 0, 0.06)
