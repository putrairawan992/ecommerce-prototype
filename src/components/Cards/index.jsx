import React from "react";
import { Card as CardAnt, Progress, Rate } from "antd";
import ButtonPlay from "../ButtonPlay";
import classNames from "classnames";
import style from "./style.sass";
import PropTypes from "prop-types";
import currencyRupiah from "../../library/currency";

export default function Cards(props) {
  const classNamesStyle = classNames.bind(style);

  const cssClassesWrapper = classNamesStyle({
    "mp-card-default": props.type === "default",
    "mp-card-collection": props.type === "collection",
    "mp-card-large": props.type === "large",
    "mp-card-small": props.type === "small"
  });

  const cardTitle = classNamesStyle({
    "mp-card-default__title": props.type === "default",
    "mp-card-collection__title": props.type === "collection",
    "mp-card-large__title": props.type === "large",
    "mp-card-small__title": props.type === "small"
  });

  const cardPrice = classNamesStyle({
    "mp-card-default__price": props.type === "default",
    "mp-card-collection__price": props.type === "collection",
    "mp-card-large__description": props.type === "large",
    "mp-card-small__description": props.type === "small"
  });

  const boxImage = classNamesStyle({
    "mp-card-default__box-image": props.type === "default",
    "mp-card-collection__box-image": props.type === "collection",
    "mp-card-large__box-image": props.type === "large",
    "mp-card-small__box-image": props.type === "small"
  });

  const cardImage = classNamesStyle({
    "mp-card-default__image": props.type === "default",
    "mp-card-collection__image": props.type === "collection",
    "mp-card-large__image": props.type === "large",
    "mp-card-small__image": props.type === "small"
  });

  return (
    <div>
      <CardAnt
        bordered={false}
        className={cssClassesWrapper}
        bodyStyle={{ padding: "8px 16px" }}
        cover={
          <div className={boxImage}>
            {props.discount && (
              <div className="mp-card-badge">{props.discount}</div>
            )}
            <img alt="example" src={props.urlImage} className={cardImage} />
            {props.showPlayButton && <ButtonPlay type="thumbnail" />}
          </div>
        }
      >
        <div className="mp-card-info">
          <div>
            {props.rateValue && (
              <div className="mp-card-stars">
                <Rate allowHalf value={props.rateValue} disabled />
                <span>({props.totalVote})</span>
              </div>
            )}
            <span className={cardTitle}>{props.title}</span>
            {props.discountPrice ? (
              <div className="card-discount">
                <span className={cardPrice}>{currencyRupiah(props.price)}</span>
                <span className="card-discount-price">
                  {currencyRupiah(props.discountPrice)}
                </span>
              </div>
            ) : (
              <span className={cardPrice}>{currencyRupiah(props.price)}</span>
            )}
          </div>
          {props.totalSold && (
            <div className="mp-card-bar">
              <div className="mp-card-bar__info">
                <span>Terjual {props.totalSold}</span>
              </div>
              <Progress
                percent={props.itemLeft}
                showInfo={false}
                strokeColor="#fb6900"
              />
            </div>
          )}
        </div>
      </CardAnt>
    </div>
  );
}

Cards.propTypes = {
  type: PropTypes.oneOf(["default", "collection", "large", "small"]),
  showPlayButton: PropTypes.bool,
  urlImage: PropTypes.string,
  title: PropTypes.string,
  price: PropTypes.number,
  discount: PropTypes.string,
  discountPrice: PropTypes.number,
  rateValue: PropTypes.number,
  totalVote: PropTypes.number,
  totalSold: PropTypes.number,
  itemLeft: PropTypes.number
};

Cards.defaultProps = {
  type: "default",
  showPlayButton: false
};
