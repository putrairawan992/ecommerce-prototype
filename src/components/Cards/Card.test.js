import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Cards from ".";

configure({ adapter: new Adapter() });

describe("Cards", () => {
  it("should show error card", () => {
    const wrapper = shallow(<Cards />);
    expect(wrapper).toMatchSnapshot();
  });
});
