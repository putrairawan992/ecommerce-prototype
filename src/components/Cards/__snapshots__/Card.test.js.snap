// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should show error card 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cards
    showPlayButton={false}
    type="default"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        bodyStyle={
          Object {
            "padding": "8px 16px",
          }
        }
        bordered={false}
        className="mp-card-default"
        cover={
          <div
            className="mp-card-default__box-image"
          >
            <img
              alt="example"
              className="mp-card-default__image"
            />
          </div>
        }
      >
        <div
          className="mp-card-info"
        >
          <div>
            <span
              className="mp-card-default__title"
            />
            <span
              className="mp-card-default__price"
            >
              Rp 0
            </span>
          </div>
        </div>
      </Card>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bodyStyle": Object {
          "padding": "8px 16px",
        },
        "bordered": false,
        "children": <div
          className="mp-card-info"
        >
          <div>
            <span
              className="mp-card-default__title"
            />
            <span
              className="mp-card-default__price"
            >
              Rp 0
            </span>
          </div>
        </div>,
        "className": "mp-card-default",
        "cover": <div
          className="mp-card-default__box-image"
        >
          <img
            alt="example"
            className="mp-card-default__image"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <span
                className="mp-card-default__title"
              />
              <span
                className="mp-card-default__price"
              >
                Rp 0
              </span>
            </div>,
            undefined,
          ],
          "className": "mp-card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <span
                  className="mp-card-default__title"
                />,
                <span
                  className="mp-card-default__price"
                >
                  Rp 0
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "mp-card-default__title",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Rp 0",
                  "className": "mp-card-default__price",
                },
                "ref": null,
                "rendered": "Rp 0",
                "type": "span",
              },
            ],
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          bodyStyle={
            Object {
              "padding": "8px 16px",
            }
          }
          bordered={false}
          className="mp-card-default"
          cover={
            <div
              className="mp-card-default__box-image"
            >
              <img
                alt="example"
                className="mp-card-default__image"
              />
            </div>
          }
        >
          <div
            className="mp-card-info"
          >
            <div>
              <span
                className="mp-card-default__title"
              />
              <span
                className="mp-card-default__price"
              >
                Rp 0
              </span>
            </div>
          </div>
        </Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bodyStyle": Object {
            "padding": "8px 16px",
          },
          "bordered": false,
          "children": <div
            className="mp-card-info"
          >
            <div>
              <span
                className="mp-card-default__title"
              />
              <span
                className="mp-card-default__price"
              >
                Rp 0
              </span>
            </div>
          </div>,
          "className": "mp-card-default",
          "cover": <div
            className="mp-card-default__box-image"
          >
            <img
              alt="example"
              className="mp-card-default__image"
            />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <span
                  className="mp-card-default__title"
                />
                <span
                  className="mp-card-default__price"
                >
                  Rp 0
                </span>
              </div>,
              undefined,
            ],
            "className": "mp-card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <span
                    className="mp-card-default__title"
                  />,
                  <span
                    className="mp-card-default__price"
                  >
                    Rp 0
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "mp-card-default__title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Rp 0",
                    "className": "mp-card-default__price",
                  },
                  "ref": null,
                  "rendered": "Rp 0",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
